unittest.sh

```diff
@@ -0,0 +1,163 @@
+    if [ $MEMSIZE -lt 16 ]
+    then
+	    echo -e "\e[1;31mTotal Memory:              $MEMSIZE GB\e[0m"
+    	echo -e "\e[1;31mMinimum RAM requirement:   16GB[WARNING]\e[0m"
+    else 
+	    echo -e "\e[1;32mTotal Memory:              $MEMSIZE GB\e[0m"
+    fi
+}
+
+#Check Total Disk space of the raw disks mentioned in nesetup.json
+check_disk_space(){
+    SIZE=0
+    for disk in $DISKLIST
+    do
+        CAPACITY=`fdisk -l | grep $disk | awk '{print $3,$4}' | head -1 | sed 's/,//g'`
+        DISKSIZE=($CAPACITY)
+        SIZE=$(($SIZE + $DISKSIZE))
+        if [ ${DISKSIZE[0]} -lt 40 ] 
+	    then
+		    echo -e "\e[1;31mThe recommended disk size: minimum 40GB[WARNING]"
+            echo -e "\e[1;31m"$disk  "                 "  $CAPACIT
+	    else
+             echo -e "\e[1;32m"$disk  "                 "  $CAPACITY
+    fi 
+    done
+    echo -e "\e[1;32mTotal Disk Space:          $SIZE GB"
+}
+
+#check if the network interface is ipv6
+is_nic_ipv6_check(){
+    ip -6 addr | grep $BROKER_INTERFACE  > /dev/null
+    EXIT_STATUS=`echo $?`
+    if [ $EXIT_STATUS -ne 0 ] 
+    then
+	    echo "The broker interface mentioned in /root/c0/nesetup.json: not an IPV6 address.[ERROR]"
+    fi
+
+    ip -6 addr | grep $SERVER_INTERFACE > /dev/null
+    EXIT_STATUS=`echo $?`
+    if [ $EXIT_STATUS -ne 0 ] 
+    then
+	    echo "The server interface mentioned in /root/c0/nesetup.json: not an IPV6 address.[ERROR]"
+    fi
+}
+
+#MTU size check for the  broker_interface and server_interface
+#mentioned in nesetup.json
+mtu_size(){
+
+    FRAME_SIZE=`ip link show dev $BROKER_INTERFACE | grep mtu | awk '{print $5}'`
+    if [ $FRAME_SIZE -ne 9000 ]
+    then
+        echo -e "\e[1;31The required frame size[MTU] for broker_interface: 9000.[ERROR]"
+    else
+        echo -e "\e[1;32mbroker_interface NIC MTU:  9000[OK]"
+    fi
+
+    FRAME_SIZE=`ip link show dev $SERVER_INTERFACE | grep mtu | awk '{print $5}'`
+    if [ $FRAME_SIZE -ne 9000 ]
+    then
+        echo -e "\e[1;31The required frame size[MTU] for server_interface: 9000.[ERROR]"
+    else
+        echo -e "\e[1;32mserver_interface NIC MTU:  9000[OK]"
+    fi
+}
+
+#check if the neadm cluster is ONLINE now
+neadm_status(){
+    NEADM_STATUS=`docker run -i -t --rm --network host nexenta/nedge-neadm /opt/neadm/neadm system status | tail -1 |  awk '{print $9}' | sed -r "s/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g"`
+
+    if [ "${NEADM_STATUS}" == 'ONLINE' ];
+    then
+	    echo -e "\e[1;32mneadm system status:       ONLINE[OK]\e[0m"
+    else
+	    echo -e "\e[1;31mThe neadm system:          NOT ONLINE[WARNING]\e[0m"
+    fi
+}
+
+#check neadm services are running such as docker ccowgw auditserv ccowserv logger networkWorker restWorke
+neadm_services(){
+
+    echo -e "\e[1;32mChecking nedge services...\e[0m"
+    for PROCESS in docker ccowgw auditserv ccowserv logger networkWorker restWorker 
+    do
+	    CHECK=$0
+    	OUTPUT=$(ps aux | grep -v grep | grep -v $CHECK | grep $PROCESS)
```

Comment:
Ideally you parse output of `/opt/nedge/nmf/nefadm status | grep online`    for  (rest network ccowserv docker ccowgws3)